packages/react-router/src/router.ts

```diff
@@ -260,9 +260,9 @@ export function createRouter<
 }
 
 export class Router<
-  TRouteTree extends AnyRoute = AnyRoute,
-  TDehydrated extends Record<string, any> = Record<string, any>,
-  TSerializedError extends Record<string, any> = Record<string, any>,
+  in out TRouteTree extends AnyRoute = AnyRoute,
```

Comment:
We can make these invariant to stop TS from doing variance checks on them. I don't think they're necessary in this case